distributed/worker.py

```diff
@@ -3412,17 +3429,22 @@ async def execute(self, key: str, *, stimulus_id: str) -> None:
 
             args2, kwargs2 = self._prepare_args_for_execution(ts, args, kwargs)
 
-            if ts.annotations is not None and "executor" in ts.annotations:
-                executor = ts.annotations["executor"]
-            else:
+            try:
+                executor = ts.annotations["executor"]  # type: ignore
+            except (TypeError, KeyError):
                 executor = "default"
```

Comment:
This is a change in behaviour that is not related to the state machine. I would prefer doing this in a separate PR.